test/integration/client/cancel-query-tests.js

```diff
@@ -0,0 +1,70 @@
+'use strict'
+
+const helper = require('./test-helper')
+const pg = helper.pg
+
+const suite = new helper.Suite()
+const Query = helper.pg.Query
+
+suite.test('cancel query interface', () => {
+  const client = new pg.Client()
+  return client.connect()
+    .then(() => {
+      const promise1 = new Promise(function(resolve, reject) {
+        const query1 = client.query(new Query('select pg_sleep(1) as sleep1'))
+        query1.on('row', function (row, result) {
+          assert(false)
+          reject();
+        })
+        query1.on('end', (err) => {
+          assert(false)
+          reject();
+        })
+        query1.on('error', (err) => {
+          assert.equal(err.code, 57014);
```

Comment:
Could define the error code above in a constant, so that it's not so much of a magic number.